### PR TITLE
fix #137; add tests; add changelog

### LINKED ISSUE #137: Self referencing foreign key cannot be used with ``fk`` pragma
```nim
type
  Foo[T] {.tableName: "Fooz".} = ref object of Model
    name*: string
    desc*: string
    # fooid*: Option[Foo[T]] # <- This works
    fooid* {.fk: Foo.}: Option[int64] # <- This doesn't works
```

This is due to : 
* the condition evaluation around ``hasCustomPragma(fk)`` which expects an integer (so Option[int] is forbidden)
* A type referencing itself through the fk pragma is incoplete and is not seen as a Model during type evaluation.

### PR DESCRIPTION
**Pull request by Clonkk**
PR Following #137

### CODE REVIEW

File: src/norm/sqlite.nim [hunk 1]
```diff
@@ -124,7 +130,7 @@ proc createTables*[T: Model](dbConn; obj: T) =
 
 # Row manipulation
 
-proc insert*[T: Model](dbConn; obj: var T, force = false, conflictPolicy = cpRaise) =
+proc insert*[T: Model](dbConn; obj: var T; force = false; conflictPolicy = cpRaise) =
```
Comment (moigagoo) [inline src/norm/sqlite.nim:133]: @PhilippMDoerner are you using an autoformatter? I'm not really against semicolons but I'd rather have one style across the codebase than two ones :-)

Could you please replace semicolons with commas?

**Review by moigagoo — APPROVED** [review]

File: src/norm/sqlite.nim (hunk 1 above)
Comment (Clonkk) [reply src/norm/sqlite.nim:133]: Ah yes, I have nimpretty set up to apply automatically. I don't know why it does that, I'll fix it

Comment (PhilippMDoerner) [reply src/norm/sqlite.nim:133]: @moigagoo not my PR :laughing:

Comment (moigagoo) [reply src/norm/sqlite.nim:133]: @PhilippMDoerner Oops, sorry! 😬

**Comment by Clonkk** [comment]
@moigagoo  Do you think it would be doable to put  a tag once this PR is done in order to specify a minimal version so we can use this feature with Nimble ?

**Review by moigagoo — APPROVED** [review]

**Comment by moigagoo** [comment]
@Clonkk here you go, released 2.3.6.

Thanks for your contributions!

**Comment by moigagoo** [comment]
@Clonkk the tests are failing. Investigating.

**Comment by Clonkk** [comment]
Oh that's weird it was ok when I ran them locally

**Comment by moigagoo** [comment]
> Out of curiosity why not execute the test on PR to branch devel ?

BTH I was 100% sure they do run on develop after PR. I think I just didn't let them finish, wanted to move the release faster.

**Comment by Clonkk** [comment]
How do you feel about running them on the PR itself ?

**Pull request merged**